src/types.ts

```diff
@@ -2900,6 +2900,12 @@ export type ImportTask = {
 };
 
 export type MessageSetType = 'latest' | 'current' | 'new';
+export type MessageSet<StreamChatGenerics extends ExtendableGenerics = DefaultGenerics> = {
+  isCurrent: boolean;
+  isLatest: boolean;
+  messages: FormatMessageResponse<StreamChatGenerics>[];
+  pagination: { hasNext?: boolean; hasPrev?: boolean };
```

Comment:
If I understand the semantics correctly:
- `true` means there's _maybe_ more to load
- `false` means there's _definitely_ nothing else to load
- `undefined` means nothing has been queried yet

Because both `true` and `undefined` mean "try querying", an easy mistake to make will be writing `if (messages.pagination.hasNext) { ... }`. Maybe we can make this type a bit more verbose:

```
pagination: { maybeHasNext: boolean; maybeHasPrev: boolean }
```

And always initialize with `{ maybeHasNext: true; maybeHasPrev: true }`

Reply:
Makes sense to keep it true. But I would keep `maybe` out of the name as it reveals our internals that can change in the future (maybe once this will be fixed on the BE side and each response will provide information about the next page availability :)).

Reply:
`hasOlder` - I am not sure whether the messages have to be always ordered by creation date. Also if we wanted to paginate other types of items, they may not be sorted by date - that is why I think it does not give us added value and the integrators should know, how they sort their items.